test/Microsoft.SqlTools.ServiceLayer.UnitTests/Utility/TestObjects.cs

```diff
@@ -158,9 +158,50 @@ public override int ExecuteNonQuery()
             throw new NotImplementedException();
         }
 
+        /// <summary>
+        /// Function to check for duplicates in Data's rows.
+        /// Returns the max duplicate count for a unique row value (int).
+        /// </summary>
         public override object ExecuteScalar()
         {
-            throw new NotImplementedException();
+            if (Data != null)
```

Comment:
I'm not super pleased with having this kind of specific logic tied here. I'd suggest creating a new class TestEditDataSqlCommand that extends TestSqlCommand and then implements this function there.